gcloud-java-contrib/gcloud-java-nio/src/main/java/com/google/gcloud/storage/contrib/nio/CloudStorageObjectAttributes.java

```diff
@@ -131,25 +145,27 @@ public FileTime lastAccessTime() {
 
   @Override
   public Object fileKey() {
-    return metadata.getFilename();
+    return name;
```

Comment:
Each of `blobId`, `id` or `mediaLink` gives you a canonical reference to the object including a generation number (version of that file).

`selfLink` give you a canonical reference to the object without generation number.

As, I think, your file system does not consider generation number (other then latest) I think you should pick the latter (`selfLink`) or Maybe creating a `BlobId` without generation number such as`BlobId.of(info.bucket, info.name)` which will make it easier later to get that blob given this key.